ilcsoft/marlinpkg.py

```diff
@@ -67,3 +67,5 @@ def postCheckDeps(self):
                 if os.path.exists(libname):
                     self.parent.module('Marlin').envpath["MARLIN_DLL"].append(libname)
                     break
+                else:
```

Comment:
Do we know now, why this sometimes seems to only work on the second try?

Reply:
This [else](https://docs.python.org/3.9/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops) should be de-dented one level
```
for ...:
    if ..:
        break
else:
    print("error: only when no 'break'")
```

Reply:
I think this function `postCheckDeps` is called before the compilation happened, so the libraries will not exist at this point. The setting for MARLIN_DLL would need to move to `checkInstall` instead?

Reply:
Except that `init_ilcsoft.sh` is written before the compilation is done...

Reply:
Ah, that would explain what we saw almost perfectly...

Reply:
Copying the code that writes the init_ilcsoft.sh file and adding `f.close()` does the trick for me.

https://github.com/iLCSoft/iLCInstall/compare/master...andresailer:iLCInstall:python3

(used for building the nighlty clicdp.cern.ch ilcsoft stacks)

Reply:
Nice. Can the necessary commits simply be cherry picked to here? Or do we need to do it manually?

Reply:
I think one would need to manually cherry-pick